### PR TITLE
routing: stop printing warning if the user passes a wrong URL

### PR DESCRIPTION
**Pull request by jd**
When a _lookup method accepts N + *remainder arguments but the number of
path components (separated by /) is < N, a TypeError is raised when calling the
object and Pecan prints a RuntimeWarning that is log. Unfortunately this just
fills up the log with no interesting information for the developer as this is
just a standard 404 error.

Change-Id: Ief1f6ce85e9416cc8180683518ef6d056b0e7f09

### CODE REVIEW

**Comment by alfredodeza** [comment]
This looks like it should be handled by a logging statement at a certain level (maybe DEBUG?).

You are right that a normal 404 will find this, but I've hit this when expecting a correct lookup with incorrect args. So I do see this as useful, and would be -1 on this approach.

**Comment by alfredodeza** [comment]
There certainly is logging throughout Pecan. In `core.py`:

    import logging
    ...
    logger = logging.getLogger(__name__)

It is also set/defined with the scaffolds to fine-tune the level of verbosity just for Pecan:

    logging = {
        'loggers': {
            'root': {'level': 'INFO', 'handlers': ['console']},
            'app_name': {'level': 'DEBUG', 'handlers': ['console']},
            'pecan': {'level': 'DEBUG', 'handlers': ['console']},
            'pecan.commands.serve': {'level': 'DEBUG', 'handlers': ['console']},
            'py.warnings': {'handlers': ['console']},
            '__force_dict__': True
        },

**Comment by jd** [comment]
Of course I only looked into `routing.py`. :)

File: pecan/routing.py [hunk 1]
```diff
@@ -170,18 +173,16 @@ def lookup_controller(obj, remainder, request=None):
 def handle_lookup_traversal(obj, args):
     try:
         result = obj(*args)
+    except TypeError as te:
+        logger.debug('Got exception calling lookup(): %s (%s)',
+                     te, te.args)
+    else:
```
Comment (alfredodeza) [inline pecan/routing.py]: there is no need for this else here unless you think you might get a `TypeError` in the next few lines? This whole block could just go below `result = obj(*args)` as it was before?

Comment (jd) [inline pecan/routing.py]: You want to log if there's a problem (`TypeError`) when calling the `obj`. If any of the other line raises a `TypeError`, it'd be a bug in Pecan. You don't want to catch that.

Yes it's unlikely the few lines will raise in `TypeError` but there's no good reason to keep them in the except block.

Comment (alfredodeza) [inline pecan/routing.py]: sounds like out of scope for what this PR is trying to do

Comment (jd) [inline pecan/routing.py]: lol, first time I hear proper Python code is "out of scope".
I've moved it in a separate commit for the sake of it.

**Comment by jd** [comment]
Actually I think the `TypeError` is also raised by a invalid return from _lookup, so that's even better than I thought. It may require a new test though.

**Comment by jd** [comment]
ping @alfredodeza ?

File: pecan/routing.py (hunk 1 above)
Comment (alfredodeza) [inline pecan/routing.py]: I didn't say this wasn't proper Python, and I wasn't trying to make a funny statement. PRs for improving the codebase are good, but can be confusing if the intention of the PR is not clear and it is trying to address other problems that are not specified in the description, which was the case here.

**Comment by alfredodeza** [comment]
This looks OK to me. @ryanpetrello do you want to double check?

**Comment by markmcclain** [comment]
LGTM
